src/macaron/malware_analyzer/pypi_heuristics/sourcecode/suspicious_setup.py

```diff
@@ -32,86 +33,86 @@ def _get_setup_source_code(self, api_client: PyPIRegistry) -> str | None:
 
         Parameters
         ----------
-            package_name (str): PyPI package's name
-            version (str): The version of the package
-            destination_dir (str): The location of the tmp
+        api_client: PyPIRegistry
+            The API client to use for source code retrieval.
 
         Returns
         -------
-            str | None: Source code.
+        str | None
+            The source code.
         """
-        # Create a temporary directory to store the downloaded source
+        sourcecode_url: str | None = api_client.get_sourcecode_url()
+        if sourcecode_url is None:
+            return None
+
+        # Create a temporary directory to store the downloaded source.
         with tempfile.TemporaryDirectory() as temp_dir:
-            sourcecode_url: str | None = api_client.get_sourcecode_url()
-            if sourcecode_url is None:
-                return None
             try:
                 response = requests.get(sourcecode_url, stream=True, timeout=40)
                 response.raise_for_status()
             except requests.exceptions.HTTPError as http_err:
                 logger.debug("HTTP error occurred: %s", http_err)
                 return None
 
-            with open(os.path.join(temp_dir, "tmp"), "wb") as f:
-                f.write(response.content)
+            if response.status_code != 200:
+                return None
 
-            files = os.listdir(temp_dir)
+            source_file = os.path.join(temp_dir, "tmp")
+            with open(source_file, "wb") as file:
+                try:
+                    for chunk in response.iter_content():
+                        file.write(chunk)
```

Comment:
I wonder why we switched from `f.write(response.content)` to usin the `iter_content()` method :thinking: ?

Reply:
As `stream=True` was already set for the `request`, which was most likely a good idea as the source code might be quite large, using `iter_content` is the suggested way to handle the received data chunks as they come in.